### PR TITLE
Make property-no-unknown ignore SCSS nested properties, fixes #1870

### LINKED ISSUE #1870: False positives with SCSS Nested Properties in property-no-unknown
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where valid CSS is being reported as an issue.

> Which rule, if any, is this issue related to?

`property-no-unknown`

> What CSS is needed to reproduce this issue?

e.g.

```css
.test {
    border: {
            style: solid;
            color: red;
    }
}
```

> What stylelint configuration is needed to reproduce this issue?

stylelint-config-standard 13.0.0

> Which version of stylelint are you using?

7.2.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Scripted via npm (`stylelint -q -s 'scss' "src/**/*.css"`)

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, SCSS nested properties.

> What did you expect to happen?

No errors raised.

> What actually happened (e.g. what warnings or errors you are getting)?

The following errors were flagged:

```shell
Unexpected unknown property "style"   property-no-unknown
```

### PR DESCRIPTION
**Pull request by epmatsw**
Fixes #1870

### CODE REVIEW

File: src/utils/isStandardSyntaxDeclaration.js [hunk 1]
```diff
@@ -6,16 +6,19 @@
  */
 export default function (decl) {
 
-  const { prop } = decl
+  const { prop, parent } = decl
 
   // Declarations belong in a declaration block
-  if (decl.parent.type === "root") { return false }
+  if (parent.type === "root") { return false }
```
Comment (evilebottnawi) [inline src/utils/isStandardSyntaxDeclaration.js]: Good catch :smile:

File: src/utils/isStandardSyntaxDeclaration.js [hunk 2]
```diff
@@ -6,16 +6,19 @@
...
 
   // SCSS var (e.g. $var: x), nested list (e.g. $list: (x)) or nested map (e.g. $map: (key:value))
   if (prop[0] === "$") { return false }
 
   // Less var (e.g. @var: x), but exclude variable interpolation (e.g. @{var})
   if (prop[0] === "@" && prop[1] !== "{") { return false }
 
+  // SCSS nested properties (e.g. border: { style: solid; color: red; })
+  if (parent.selector[parent.selector.length - 1] === ":") { return false }
```
Comment (evilebottnawi) [inline src/utils/isStandardSyntaxDeclaration.js]: Can you add tests on this line in `isStandardSyntaxDeclaration-test.js`

Comment (epmatsw) [inline src/utils/isStandardSyntaxDeclaration.js]: Will do! Don't know how I didn't see that file.

File: src/rules/property-no-unknown/index.js [hunk 3]
```diff
@@ -37,6 +38,7 @@ export default function (actual, options) {
       if (!shouldCheckPrefixed && vendor.prefix(prop)) { return }
 
       if (!isStandardSyntaxProperty(prop)) { return }
+      if (!isStandardSyntaxDeclaration(decl)) { return }
       if (isCustomProperty(prop)) { return }
```
Comment (evilebottnawi) [inline src/rules/property-no-unknown/index.js]: I think we should move up `isStandard*` and `isCustomProperty` than `if (!shouldCheckPrefixed && vendor.prefix(prop)) { return }`. Why check prefixes on non standard properties? It is not you small mistake, but we can do this in this PR.

**Comment by evilebottnawi** [comment]
@epmatsw Good work, some comments and LGTM

File: src/utils/isStandardSyntaxDeclaration.js (hunk 2 above)
Comment (evilebottnawi) [inline src/utils/isStandardSyntaxDeclaration.js]: @epmatsw https://github.com/stylelint/stylelint/blob/master/src/utils/__tests__/isStandardSyntaxDeclaration-test.js

File: src/rules/property-no-unknown/__tests__/index.js [hunk 4]
```diff
@@ -66,6 +66,9 @@ testRule(rule, {
   }, {
     code: ".foo { #{$prop}: black; }",
     description: "ignore property interpolation",
+  }, {
```
Comment (jeddy3) [inline src/rules/property-no-unknown/__tests__/index.js]: Could we get *reject* test for an unknown property *within a [custom property set](https://tabatkins.github.io/specs/css-apply-rule/#defining)* please. e.g.

```js
{
  code: ".foo { --custom-property-set: { colr: blue; } }",
  message: messages.rejected("colr"),
  line: 1,
  column: 32,
}
```

Likewise you'll want to check that declarations within custom property sets are *accepted* within `isStandardSyntaxDeclaration-test.js`.

I suspect the code will need to be tweaked to make these tests pass as the conditional currently doesn't seem to distinguish between nested properties and custom property sets.

**Comment by jeddy3** [comment]
@epmatsw Thanks for taking another stab. I've made a request around custom property sets. Otherwise, looks good to me :)

**Comment by epmatsw** [comment]
@jeddy3 @evilebottnawi I think I got all the comments. Let me know if there's more tests or anything needed, you all are clearly much more familiar with obscure CSS syntax than I am hahaha

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,7 @@
 -   Fixed: `block-no-single-line` now ignores empty blocks.
 -   Fixed: `--ignore-path` and `--report-needless-disables` no longer fails when used together.
 -   Fixed: the `indentation` rule now correctly handles `_` hacks on property names.
+-   Fixed: the `property-no-unknown` rule now ignores scss nested properties.
```
Comment (davidtheclark) [inline CHANGELOG.md]: Please capitalize SCSS.

**Comment by jeddy3** [comment]
@epmatsw Thanks for addressing my requests.

Besides @davidtheclark's minor request, this looks ready to merge.

File: CHANGELOG.md (hunk 5 above)
Comment (epmatsw) [inline CHANGELOG.md]: Done! :)

**Comment by epmatsw** [comment]
@jeddy3 @davidtheclark changelog updated :)

**Comment by jeddy3** [comment]
LGTM

**Pull request merged**